include/openssl/cipher.h

```diff
@@ -643,6 +643,12 @@ BSSL_NAMESPACE_END
 
 #endif
 
+/*
+ * The maximum permitted number of cipher blocks per data unit in XTS mode.
+ * Reference IEEE Std 1619-2018.
+ */
+#define XTS_MAX_BLOCKS_PER_DATA_UNIT            (1<<20)
```

Comment:
Can we move to an internal header file?

Reply:
I had first put it in an internal header file, then decided to move it here when I added the errors below, especially for the user to know the maximum allowed input size.

Reply:
Sure, but the more defines we expose publicly the harder agility gets in the future. I.e. simply renaming it is breaking.

Leave it to you, but I lean towards not exposing it.

Reply:
Good point. I think I opted to expose it as is done in OpenSSL in [include/crypto/modes.h](https://github.com/openssl/openssl/blob/8020d79b4033400d0ef659a361c05b6902944042/include/crypto/modes.h#L140-L144).

Reply:
I don't think header files in `include/crypto` are intended for application usage. The content is nominally opaque.

Reply:
Moved it inside in https://github.com/awslabs/aws-lc/pull/400/commits/f57cfb2d162155a584374a89b7f2bada09f4d3b2. Thanks.